app/scripts/modules/kubernetes/src/v2/manifest/delete/delete.controller.ts

```diff
@@ -0,0 +1,86 @@
+import { copy, IController, module } from 'angular'
+import { IModalServiceInstance } from 'angular-ui-bootstrap';
+
+import {
+  Application,
+  MANIFEST_WRITER,
+  ManifestWriter,
+  TASK_MONITOR_BUILDER,
+  TaskMonitor,
+  TaskMonitorBuilder
+} from '@spinnaker/core';
+import { IManifestCoordinates } from '../IManifestCoordinates';
+
+interface IDeleteCommand {
+  name: string;
+  location: string;
+  account: string;
+  reason: string;
+  options: IDeleteOptions;
+}
+
+interface IDeleteOptions {
+  gracePeriodSeconds?: number;
+  cascading: boolean;
+}
+
+class KubernetesManifestDeleteController implements IController {
+  public taskMonitor: TaskMonitor;
+  public command: IDeleteCommand;
+  public verification = {
+    verified: false
+  };
+
+  constructor(coordinates: IManifestCoordinates,
+              taskMonitorBuilder: TaskMonitorBuilder,
+              private $uibModalInstance: IModalServiceInstance,
+              private manifestWriter: ManifestWriter,
+              private application: Application) {
+    'ngInject';
+
+    this.taskMonitor = taskMonitorBuilder.buildTaskMonitor({
+      title: `Deleting ${coordinates.name} in ${coordinates.namespace}`,
+      application: application,
+      modalInstance: $uibModalInstance,
+    });
+
+    this.command = {
+      name: coordinates.name,
+      location: coordinates.namespace,
+      account: coordinates.account,
+      reason: null,
+      options: {
+        cascading: true
+      }
+    };
+  }
+
+  public isValid(): boolean {
+    return this.verification.verified &&
+      this.command.options.gracePeriodSeconds == null || this.command.options.gracePeriodSeconds >= 0;
```

Comment:
Use triple equals unless you’re checking for both null and undefined here... which is maybe exactly what you’re doing.